R-package/R/lgb.prepare_rules.R

```diff
@@ -50,6 +53,12 @@
 #' @export
 lgb.prepare_rules <- function(data, rules = NULL) {
 
+  .Deprecated(
+    new = "lgb.prepare2"
+    , package = "lightgbm"
+    , msg = "lgb.prepare is deprecated and will be removed in LightGBM 3.0.0. lgb.prepare2() is preferred."
```

Comment:
`msg` should read: `lgb.prepare_rules is deprecated and will be removed in LightGBM 3.0.0. lgb.prepare_rules2() is preferred.` (change accordingly the other field).

Reply:
Maybe we should also rename `prepare2` into `prepare` with 3.0 release? I believe `2` without `1` makes no sense.

Reply:
in my opinion it is not worth breaking existing code just for name consistency

Reply:
I think the name should not be changed. However, maybe we can set an alias for the name consistency, and remove the old one after the next release.

Reply:
@jameslamb does `prepare2` do the same thing as `prepare`? If yes, I agree with @StrikerRUS .

Reply:
They don't do the exact same thing.

`lgb.prepare()` converts columns that are `character` or `factor` to `numeric`: https://lightgbm.readthedocs.io/en/latest/R/reference/lgb.prepare.html

`lgb.prepare2()` converts columns that are `character` or `factor` to `integer`: https://lightgbm.readthedocs.io/en/latest/R/reference/lgb.prepare2.html

Since LightGBM can accept integer input, I think we should only support the one that produces integers, since it is a more efficient representation.

Both functions achieve the same goal: take an R data frame that is not suitable as a training dataset and convert it to one that is.

This is why I think we should remove `lgb.prepare()`, but not change the names. Or, if you are worried about the `2`, we should pick a new and completely different name, like `lgb.convert()` or something. I believe that breaking changes in a new major version should be loud.

Reply:
I'm +1 for deprecating both versions and the completely new name in 3.0.

Reply:
when I opened this PR in May, the intention was to truly "deprecate" these, meaning "raise a warning saying they'll be removed in 3.0.0".

At this point, I think it's too late for that. I think we should just remove them and make that one of the breaking changes in the 3.0.0 release. 

So specifically:

1. Remove `lgb.prepare()` and `lgb.prepare_rules()`
2. Rename `lgb.prepare2()` to `lgb.convert()`
3. Rename `lgb.prepare_rules2()` to `lgb.convert_with_rules()`

@guolinke what do you think?

Reply:
@jameslamb looks good to me.

Reply:
ok I will change this PR to do that then, thank you!

Reply:
ok this has been changed! It's ready for another review